### DISCUSSION TITLE
Query commands for complete HA energy monitor

### DISCUSSION THREAD

**Discussion by RoySalisbury** [jblance/mpp-solar#315] [General]
Just curious, what commands are people using to output detailed data for Home Assistant integration.  

I'm assuming your running as a daemon and using a config file, so posting that would be helpful.

Roy

**Comment by Saentist** [comment]
**`QPIGS`** as minimum

**Reply by RoySalisbury** [reply]
Here is mine BTW ....

```
[SETUP]
pause=5
mqtt_broker=****
mqtt_user=****
mqtt_pass=****

[EG46500-01]
protocol=PI30MAX
port=/dev/ttyUSB0
command=QPIGS#QPIGS2
outputs=hassd_mqtt,screen
tag=EG46500_01
```


Unfortunately, PI30MAX does not seem to work for hassd_mqtt ..  That and not all the necessary meta data is present for HA to incorporate the PV and inverters into the energy dashboard.

**Reply by Saentist** [reply]
Remove `screen` from service config.
Service run in background.

**Reply by RoySalisbury** [reply]
`screen` was added just so I could see what was going on (e.g, debug).

**Reply by RoySalisbury** [reply]
And the missing meta data is not specifically an issue with mpp-solar.  Its just not something the basic commands will give you (accumulated daily values). 

You would need to be able to run `QEDyyyymmdd` dynamically (replacing the date at runtime)

**Comment by jblance** [comment]
How doesnt pi30max work with hassd_mqtt?
It's the setup I use (well I actually use the new powermon command but o
was using it with mppsolar command successfully)

**Comment by RoySalisbury** [comment]
I may have misspoke based on the output displayed.  

Here is the output when the config specifies PI30MAX as the protocol.

```
Getting results from device: mppsolar device - name: EG46500-01, port: <mppsolar.inout.serialio.SerialIO object at 0xffffb78a39a0>, protocol: PI30 protocol handler for LV6048MAX and similar inverters for command: QPIGS2, tag: EG46500_01, outputs: hassd_mqtt,screen
Command: Unknown command - No description found
--------------------------------------------------------------------------------
Parameter          Value          	Unit
pv2_input_current  0.0            	A   	{'icon': 'mdi:solar-power', 'device-class': 'current'}
pv2_input_voltage  0.0            	V   	{'icon': 'mdi:solar-power', 'device-class': 'voltage'}
pv2_charging_power 0              	W   	{'icon': 'mdi:solar-power', 'device-class': 'power'}
```

Notice that it says its using PI30 as the protocol handler, and the command is unknown.
